### PR TITLE
Xav/rbacmerged

### PR DESCRIPTION
**Pull request by xrajesh**
**Related Issue:**  stolostron/backlog#<ISSUE_NUMBER>

### CODE REVIEW

File: pkg/rbac/cache.go [hunk 1]
```diff
@@ -25,12 +22,9 @@ type Cache struct {
 	// Clients to external APIs.
 	// Defining these here allow the tests to replace with a mock client.
 	authnClient  authnv1.AuthenticationV1Interface
-	authzClient  authzv1.AuthorizationV1Interface
 	corev1Client corev1.CoreV1Interface
-
-	kubeClient kubernetes.Interface
```
Comment (Anxhela21) [inline pkg/rbac/cache.go:31]: Why do we remove the kubeClient from cache?

File: pkg/rbac/userData.go [hunk 2]
```diff
@@ -18,41 +18,114 @@ import (
...
+	if err == nil {
+		klog.V(5).Info("No errors on namespacedresources not present ....Computing now", user)
+		userData, err = user.getClusterScopedResources(cache, ctx, clientToken)
+	}
 	return userData, err
 
 }
 
+func userCacheValid(user *userData) bool {
+	if (time.Now().Before(user.csrUpdatedAt.Add(time.Duration(config.Cfg.UserCacheTTL) * time.Millisecond))) &&
```
Comment (Anxhela21) [inline pkg/rbac/userData.go:66]: Can we have a comment here or update the comments in user cache clarifying that these `updatedAt` fields are the time when resources were authorized last.

File: pkg/rbac/cache.go (hunk 1 above)
Comment (xrajesh) [reply pkg/rbac/cache.go:31]: It was not getting used.

File: pkg/rbac/userData.go [hunk 3]
```diff
@@ -18,41 +18,113 @@ import (
...
+			user.authzClient = authzClient
+		}
 	}
-
 	userData, err := user.getNamespacedResources(cache, ctx, clientToken)
+
+	// Get cluster scoped resources for the user
+	// TO DO : Make this parallel operation
+	if err == nil {
+		klog.V(5).Info("No errors on namespacedresources not present ....Computing now", user)
```
Comment (Anxhela21) [inline pkg/rbac/userData.go:57]:
```suggestion
		klog.V(5).Info("No errors on namespacedresources present ....Computing now", user)
```

Comment (Anxhela21) [reply pkg/rbac/userData.go:57]: Also, do we need to log the user info? Maybe we can but at a higher verbosity?
```suggestion
		klog.V(6).Info("No errors on namespacedresources present ....Computing now", user)
```

Comment (xrajesh) [reply pkg/rbac/userData.go]: If we want to log that , as well see the user too right ? Is this high priority - or can hold @Anxhela21

**Review by SherinV — APPROVED** [review]

**Pull request merged**